### DISCUSSION TITLE
[DEV] Content progression tool (season switch)

### DISCUSSION THREAD

**Discussion by FrancescoBorzi** [azerothcore/azerothcore-wotlk#3898] [Ideas]
AzerothCore supports the 3.3.5a client and game content, however it's quite common to change the DB in order to close/open contents gradually and create some kind of progression. For example, some servers start with Naxxramas, then they change season and open Ulduar, etc... PvP vendors should be updated too.

I have the feeling that all people doing this are working over and over on the same thing, which is a waste of time.

So why don't we create a repository with all the SQL code needed to switch from one season to another?

**Comment by BarbzYHOOL** [comment]
Yes and they all do it the wrong way.

The best way is to create game events like the arena seasons, but it's more work but it's clean and "professional", just a hint for others

**Comment by Yehonal** [comment]
No, it's not enough, because there are some "UPDATES" queries on things
that are in-game...it's not just a matter of stuff that are spawned or
not.

The easiest solution is a collection of "migration queries" with which you
can change the version of the database....it's still a non-super-blizzlike
solution but doable.

In the end, also the blizzard started the official new classic experience
with the features of 1.12 client but the content of 1.1.0 I guess. So it's
a reasonable way to go.

If you think about, no one would play with old non-fixed features (think
about all bugs of 3.0.2 that have been fixed for example), but everyone
want to revive the progression and the content, the important thing is that
can still be called Wotlk.

That's why I think the database migration are the best way

**Reply by Maurowin** [reply]
Warmane did it with Events methode (like Brewfest or something else), like that no requiers queries nand restart server.

**Comment by Yehonal** [comment]
However, with the project ChromieCraft we will work on it

**Comment by kaytotes** [comment]
@Yehonal Has progress been made on this? I've been pondering this since I first saw this discussion quite some time ago. 

I have a branch in a private repo where I started working on basically porting in how Lights Hope did it with various tweaks. 

Currently covering (roughly i'll add):

- Config option for choosing emulated patch
- Client Cache busting based on patch version.
- Creature Templates
- Creature Loading (and spawning).
- Forbidden Items (for limiting drops based on patch or removing them from inventories etc)
- Creature Equip
- Various loot templates eg creature_loot_template, milling_loot_template, item_loot_template and so on.
- Quest starter and ender.
- BG Templates (eg adjusting the number of players allowed in)
- Game events

The code is from July 2020 so I'd likely need to re-work a good chunk and of course is missing huge bits that would be required but i'd be happy to do so. 

To be blunt i'm not experienced with all the required tables we'd need for this or the Azeroth-Core codebase deeply but willing to learn more. 

The reason i'd like this in Azeroth-Core is actually from the perspective of a custom content developer. Even with just implementing the above in my small team we noticed a massive speed up of development because it allowed us to work on multiple "patches" at once.
